modular_skyrat/modules/microfusion/code/microfusion_gun_attachments.dm

```diff
@@ -0,0 +1,392 @@
+	desc = "Experimental technology that inverts the central phase emitter causing the wave frequency to shift into X-ray. CAUTION: Phase emitter heats up very quickly."
+	icon_state = "attachment_xray"
+	slot = GUN_SLOT_BARREL
+	attachment_overlay_icon_state = "attachment_xray"
+	heat_addition = 90
+	power_usage = 50
+
+/obj/item/microfusion_gun_attachment/xray/run_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.fire_sound = 'modular_skyrat/modules/microfusion/sound/incinerate.ogg'
+
+/obj/item/microfusion_gun_attachment/xray/remove_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.fire_sound = microfusion_gun.chambered?.fire_sound
+
+/obj/item/microfusion_gun_attachment/xray/process_fire(obj/item/gun/microfusion/microfusion_gun, obj/item/ammo_casing/chambered)
+	. = ..()
+	chambered.loaded_projectile.icon_state = "laser_greyscale"
+	chambered.loaded_projectile.color = COLOR_GREEN
+	chambered.loaded_projectile.light_color = COLOR_GREEN
+	chambered.loaded_projectile.projectile_piercing = PASSCLOSEDTURF|PASSGRILLE|PASSGLASS
+
+/*
+GRIP ATTACHMENT
+
+Greatly reduces recoil and spread.
+*/
+/obj/item/microfusion_gun_attachment/grip
+	name = "grip attachment"
+	desc = "A simple grip that increases accuracy."
+	icon_state = "attachment_grip"
+	attachment_overlay_icon_state = "attachment_grip"
+	slot = GUN_SLOT_UNDERBARREL
+	/// How much recoil are we removing?
+	var/recoil_to_remove = 1
+	/// How much spread are we removing?
+	var/spread_to_remove = 10
+
+/obj/item/microfusion_gun_attachment/grip/run_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.recoil -= recoil_to_remove
+	microfusion_gun.spread -= spread_to_remove
+
+/obj/item/microfusion_gun_attachment/grip/remove_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.recoil += recoil_to_remove
+	microfusion_gun.spread += spread_to_remove
+
+/*
+HEATSINK ATTACHMENT
+
+"Greatly increases the phase emitter cooling rate."
+*/
+/obj/item/microfusion_gun_attachment/heatsink
+	name = "phase emitter heatsink"
+	desc = "Greatly increases the phase emitter cooling rate."
+	icon_state = "attachment_heatsink"
+	attachment_overlay_icon_state = "attachment_heatsink"
+	slot = GUN_SLOT_UNDERBARREL
+	/// Coolant bonus
+	var/cooling_rate_increase = 50
+
+/obj/item/microfusion_gun_attachment/heatsink/run_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.heat_dissipation_bonus += cooling_rate_increase
+
+/obj/item/microfusion_gun_attachment/heatsink/remove_attachment(obj/item/gun/microfusion/microfusion_gun)
+	. = ..()
+	microfusion_gun.heat_dissipation_bonus -= cooling_rate_increase
+
+/*
+UNDERCHARGER ATTACHMENT
+
+Massively decreases the output beam of the phase emitter.
+Converts shots to STAMNINA damage.
+*/
+/obj/item/microfusion_gun_attachment/undercharger
+	name = "phase emitter undercharger"
+	desc = "Inverts the output beam of the phase emitter."
```

Comment:
desc = "An underbarrel system hooked to the phase emitter, this allows the weapon to also fire an electron bolt, producing a short-lived underpowered electric charge capable of stunning targets. These shots are less demanding on the weapon, leading to an increase in cooling rate."